src/compiler/core.ts

```diff
@@ -9,6 +9,20 @@ namespace ts {
     export const version = `${versionMajorMinor}.0`;
 }
 
+namespace ts {
+    export function isExternalModuleNameRelative(moduleName: string): boolean {
+        // TypeScript 1.0 spec (April 2014): 11.2.1
+        // An external module name is "relative" if the first term is "." or "..".
+        // Update: We also consider a path like `C:\foo.ts` "relative" because we do not search for it in `node_modules` or treat it as an ambient module.
+        return pathIsRelative(moduleName) || isRootedDiskPath(moduleName);
+    }
+
+    /** @deprecated Use `!isExternalModuleNameRelative(moduleName)` instead. */
+    export function moduleHasNonRelativeName(moduleName: string): boolean {

```

Comment:
It wasn't internal all the time. tslint uses this function and it worked before the release of 2.5.0 https://github.com/palantir/tslint/blob/master/src/rules/noSubmoduleImportsRule.ts#L84

Since `moduleHasNonRelativeName` is deprecated, I'm ok with only exposing `isExternalModuleNameRelative` if you want.